### ISSUE TITLE
Ensure compatibility with WordPress 5.4

### ISSUE THREAD

**Issue by GlennMartin1** [woocommerce/storefront#1287]
## WordPress 5.4 is schedule for release this month.
(per https://make.wordpress.org/core/5-4/)

## WordPress 5.4 introduces breaking changes:
https://make.wordpress.org/core/2020/03/03/wordpress-5-4-field-guide/
https://wptavern.com/preparing-for-wordpress-5-4-changes-theme-and-plugin-developers-should-know-about

Storefront should be reviewed and tested for WordPress 5.4, prior to its release.

**Comment by nielslange** [comment]
**Things to check:**

- [ ] Calendar markup → https://make.wordpress.org/core/2020/02/12/changes-related-to-calendar-widget-markup-in-wordpress-5-4/
- [ ] New nav menu hooks → https://make.wordpress.org/core/2020/02/25/wordpress-5-4-introduces-new-hooks-to-add-custom-fields-to-menu-items/
- [ ] apply_shortcodes() alias function → https://make.wordpress.org/core/2020/02/13/wordpress-5-4-introduces-apply-shortcodes-as-an-alias-for-do-shortcode/

**Comment by haszari** [comment]
Added conservative tags to this issue (enhancement, high priority). If there are specific breakages due to this we can upgrade to `Type: Bug` (etc) as needed.

**Comment by Aljullu** [comment]
Thanks for the heads up @GlennMartin1 and the links @nielslange. I did a quick review and that's what I found out:
- Social icons block: seems to work ok without any changes. :heavy_check_mark: 
- Buttons block: seems to work ok without any changes. :heavy_check_mark: 
- Calendar markup: I created a PR (#1315) and an issue (#1314). Besides that, I think everything else is ok.
- New nav menu hooks: I don't think we need to do anything here.
- `apply_shortcodes()` as an alias for `do_shortcode()`: I'm not sure we want to do this change right now for backwards compatibility. Thoughts?
- `editor-` classes renamed to `block-editor-`: I created a PR (#1316).

If nothing else pops up, I think it should be fine to close this issue and open follow-ups if there is anything we missed.

**Comment by GlennMartin1** [comment]
Thanks much! Any ETA on the next release?

**Comment by nielslange** [comment]
> apply_shortcodes() as an alias for do_shortcode(): I'm not sure we want to do this change right now for backwards compatibility. Thoughts?

Can't we add backwards compatibility to Storefront? E.g. if a user if using `< WordPress 5.4` then `apply_shortcodes()` would act as a wrapper. Happy to work on a PR if others think that this is a good idea.

**Comment by Aljullu** [comment]
> Any ETA on the next release?

@GlennMartin1 We would like to release it in parallel with WooCommerce 4.1, expected for the first week of May. We will release a beta version prior to that to give some testing time to the community. Of course those are tentative dates. :slightly_smiling_face: 

> Can't we add backwards compatibility to Storefront? E.g. if a user if using `< WordPress 5.4` then `apply_shortcodes()` would act as a wrapper. Happy to work on a PR if others think that this is a good idea. 

@nielslange Sure, as long as it's compatible with new and old WP versions it should be good to get it merged. I see Storefront is using `storefront_do_shortcode()` instead of the default `do_shortcode()`. [README says it's because of performance reasons](https://github.com/woocommerce/storefront/blob/master/README.txt#L555), I'm not sure if they still apply, but we would need to research that as well.

**Comment by nielslange** [comment]
> @nielslange Sure, as long as it's compatible with new and old WP versions it should be good to get it merged. I see Storefront is using storefront_do_shortcode() instead of the default do_shortcode(). README says it's because of performance reasons, I'm not sure if they still apply, but we would need to research that as well.

@Aljullu That's the idea! WP 5.4+ users will use the new shortcode, other the old one. I have to look up if I'm able to implement it though.

**Comment by nielslange** [comment]
@Aljullu The needed could _"would"_ be:

```
if ( ! function_exists ( 'apply_shortcodes' ) ) {
	function apply_shortcodes( $content, $ignore_html = false ) {
		do_shortcode( $content, $ignore_html = false );
	}
}
```

_"Would"_, because Storefront is no longer using `do_shortcode` since Storefront 1.4.6. Therefore we can tick off this issue. 😁

**Comment by GlennMartin1** [comment]
I'm not expert here, but...

Do the Storefront child-themes also need updates, particularly thinking of the classnames?

**Comment by Aljullu** [comment]
> Therefore we can tick off this issue. 

Yeah... that's what I think as well, we don't need to do anything here. Thanks for checking. :+1: 

> Do the Storefront child-themes also need updates, particularly thinking of the classnames?

They will all benefit from the work we do in the base theme, so they should only worry about anything  they added/rewrote on top of Storefront. In case you see a bug, I think the way to proceed is to contact support if it's a paid theme or report the bug directly to the repo if it's open.

**Comment by Aljullu** [comment]
I will proceed closing this issue, we can create follow-ups if there is something specific that still needs to be addressed. Thank you again for taking the time to report it. :slightly_smiling_face:

**Comment by haszari** [comment]
Thanks for testing to confirm stuff and spinning off PRs for necessary follow up @Aljullu .

I think this is a priority for 2.5.6 so I've tweaked the titles & priority of the PRs. I'll test and review soon :)

**Comment by Aljullu** [comment]
Several WP 5.4 compatibility improvements will be included in Storefront 2.5.6, which is planned to be released in a couple of weeks. If you want, you can already try the Release Candidate:
https://github.com/woocommerce/storefront/releases/tag/version%2F2.5.6-rc.1
We did a lot of internal testing, but if you find an issue, we would appreciate if you could report it.
